### PR TITLE
Remove All Segments

### PR DESCRIPTION
**Pull request by Maithem**
trimPrefix was only removing segments that it held in memory, so
when the logunit restarts before a compact operation happens, some
of the segments would not be deleted. This can happen on trimmed
segments that are never loaded into memory after restarting the
logunit.

### CODE REVIEW

File: infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java [hunk 1]
```diff
@@ -387,26 +387,29 @@ private void trimPrefix() {
...
                 sh.close();
+                writeChannels.remove(sh.getFileName());
+            }
+        }
 
-                File logFile = new File(sh.getFileName());
-                File trimmedFile = new File(getTrimmedFilePath(sh.getFileName()));
-                File pendingFile = new File(getPendingTrimsFilePath(sh.getFileName()));
+        File dir = new File(logDir);
+        FileFilter fileFilter = new FileFilter() {
```
Comment (corfu-repo) [inline infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java]: MAJOR Make this anonymous inner class a lambda [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)

**Comment by corfu-repo** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java [hunk 2]
```diff
@@ -387,26 +387,34 @@ private void trimPrefix() {
...
```
Comment (corfu-repo) [inline infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java]: MAJOR Make this anonymous inner class a lambda [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)

**Comment by corfu-repo** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by corfudb-performance** [comment]
Results automatically generated by _CorfuDB Benchmark Framework_ to assess the performance of this pull request for commit **67c895a**.

*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 10 threads, Disk mode
 
 An interactive dashboard with Pull Request Performance Metrics for ALL cluster types and numbers of threads in run, is available at:
 [Pull Request #892 Graphs ](https://snapshot.raintank.io/dashboard/snapshot/oviOBavu01vlUfm7ZScm824rDrSdanzf)

**Pull request merged**